hazelcast/src/main/java/com/hazelcast/topic/impl/reliable/MessageRunner.java

```diff
@@ -93,14 +92,24 @@ public void next() {
 
     @Override
     public void accept(ReadResultSet<ReliableTopicMessage> result, Throwable throwable) {
+        if (cancelled) {
+            return;
+        }
+
         if (throwable == null) {
             // we process all messages in batch. So we don't release the thread and reschedule ourselves;
             // but we'll process whatever was received in 1 go.
-            for (Object item : result) {
-                ReliableTopicMessage message = (ReliableTopicMessage) item;
+            for (int i = 0; i < result.size(); i++) {
+                ReliableTopicMessage message = result.get(i);
 
-                if (cancelled) {
-                    return;
+                long actualSequence = result.getSequence(i);
+                if (actualSequence != sequence) {
+                    if (handleSilentSequenceJump(actualSequence)) {
+                        sequence = actualSequence;
+                    } else {
+                        cancel();
```

Comment:
How about if we cancel and throw the `StaleSequenceException` back to the user if the listener is not loss tolerant? Now, we just cancel which I guess stops the listener silently without notifying the user.

Reply:
Can pointed out that this is not possible and we don't throw anything back to the caller even for the current implementation but instead handle the exception in the runner. So ignore this.